embulk-input-s3/src/test/java/org/embulk/input/s3/TestS3InputStreamReopener.java

```diff
@@ -89,6 +90,28 @@ public void reopenS3FileByReopener_on_retry_gave_up_should_throw_original_except
         opener.reopen(0, new RuntimeException());
     }
 
+    @Test(expected = AmazonClientException.class)
+    public void reopenS3FileByReopener_on_retry_always_throw_exception()
+            throws Exception
+    {
+        { // always failed call with 2 retries
+            S3Object s3Object = mock(S3Object.class);
+            doReturn(s3Object).when(client).getObject(any(GetObjectRequest.class));
```

Comment:
If it always throw, why `doReturn` is necessary? if it is to avoid a NullPointerException for the later call to `getObjectContent()`, this is too much white boxing. Any reason to not just throw on `doGetObject`?

Reply:
Fixed.